src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/BreakpointEvidence.java

```diff
@@ -101,6 +102,102 @@ public String stringRep(final ReadMetadata readMetadata, final int minEvidenceMa
         return location.toString() + "\t" + weight + "\t" + this.getClass().getSimpleName() + "\t" + dtString;
     }
 
+    /**
+     * Returns BreakpointEvidence constructed from string representation. Used to reconstruct BreakpointEvidence for
+     * unit tests. It is intende for stringRep() to be an inverse of this function, but not the other way around. i.e.
+     *      fromStringRep(strRep, readMetadata).stringRep(readMetadata, minEvidenceMapQ) == strRep
+     * but it may be the case that
+     *      fromStringRep(evidence.stringRep(readMetadata, minEvidenceMapQ), readMetadata) != evidence
+     */
+    @VisibleForTesting
+    static BreakpointEvidence fromStringRep(final String strRep, final ReadMetadata readMetadata) {
+        final String[] words = strRep.split("\t");
+
+        final SVInterval location = locationFromStringRep(words[0]).getInterval();
+
+        final int weight = Integer.parseInt(words[1]);
+
+        final String evidenceType = words[2];
+        if(evidenceType.equals("TemplateSizeAnomaly")) {
+            final int readCount = Integer.parseInt(words[4]);
+            return new TemplateSizeAnomaly(location, weight, readCount);
+        } else {
+            final List<StrandedInterval> distalTargets = words[3].isEmpty() ? new ArrayList<>()
+                : Arrays.stream(words[3].split(";")).map(BreakpointEvidence::locationFromStringRep)
+                    .collect(Collectors.toList());
+            if(distalTargets.size() > 1) {
+                throw new IllegalArgumentException("BreakpointEvidence must have 0 or 1 distal targets");
+            }
+            final String[] templateParts = words[4].split("/");
```

Comment:
The different types of BreakpointEvidence have different properties, and are therefore reconstructed from dumped strings differently. TemplateSizeAnomolies are particularly different from the others, since they are not based on individual reads. Similarly when constructing feature vectors, their features will differ depending on evidence type.

Reply:
I don't see this in the new commits so it's probably moot, but if something like this is really necessary I think it would be safer to move responsibility for parsing into the thing being parsed.